src/main/engine/engine.ts

```diff
@@ -43,6 +43,9 @@ const _tapTempoEngine = new TapTempoEngine()
 function _tapTempo() {
   _tapTempoEngine.tap((newBpm) => {
     _nodeLink.setTempo(newBpm)
+  }, (newPhase) => {
+    const info = _nodeLink.getSessionInfoCurrent();
+    _nodeLink.forceBeat(info.beats - info.phase + newPhase);
```

Comment:
We might want to experiment whether forcing the beat is too aggressive and `requestBeat` would be better, though a smooth remapping of the beat would also defeat the purpose here a bit (maybe force the first beat and then request the following ones?)

Reply:
This seems to prevent the modulation from jumping after the first beat, so I've implemented that for now